azure_sdk/spec/azure_sdk_spec.rb

```diff
@@ -0,0 +1,125 @@
+# encoding: utf-8
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+require_relative 'spec_helper'
+
+describe 'Azure Sdk' do
+  describe 'Management plane' do
+    before(:each) do
+      @tenant_id = '22222222-2222-2222-2222-222222222222'
+      @client_id = '22222222-2222-2222-2222-222222222222'
+      @client_secret = 'VERY_SECRET'
+      @subscription_id = '22222222-2222-2222-2222-222222222222'
+      @active_directory_settings = MsRestAzure::ActiveDirectoryServiceSettings.get_azure_china_settings
+      @minimum_valid_options = {
+          :tenant_id => @tenant_id,
+          :client_id => @client_id,
+          :client_secret => @client_secret,
+          :subscription_id => @subscription_id,
+          :active_directory_settings => @active_directory_settings
+      }
+    end
+
+    after(:each) do
+      Azure::ARM.client.reset!
+    end
+
+    it 'should initialize with client constructor' do
+      arm_client = Azure::ARM.client(@minimum_valid_options)
+      verify_client_defaults(arm_client)
+      verify_management_clients(arm_client)
+    end
+
+    it 'should initialize by defaulting to ENV' do
+      arm_client = Azure::ARM.client({})
+      expect(arm_client).not_to be_nil
+      expect(arm_client).to be_instance_of(Azure::ARM::Client)
+      expect(arm_client.tenant_id).to eq(ENV['AZURE_TENANT_ID'])
+      expect(arm_client.client_id).to eq(ENV['AZURE_CLIENT_ID'])
+      expect(arm_client.client_secret).to eq(ENV['AZURE_CLIENT_SECRET'])
+      expect(arm_client.subscription_id).to eq(ENV['AZURE_SUBSCRIPTION_ID'])
+      expect(arm_client.credentials).to  be_kind_of(MsRest::ServiceClientCredentials)
+      verify_management_clients(arm_client)
+    end
+
+    it 'should initialize with credentials and subscription id' do
+      Azure::ARM.credentials = MsRest::TokenCredentials.new(MsRestAzure::ApplicationTokenProvider.new(
+          @tenant_id, @client_id, @client_secret, @active_directory_settings))
+      Azure::ARM.subscription_id = @subscription_id
+
+      verify_management_clients(Azure::ARM)
```

Comment:
Sure

---
In reply to: [76344795](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344795) [](ancestors = 76344795)